### ISSUE TITLE
[accessibility issue] Landmarks are missing and incorrectly marked up

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#79521]
**Steps to reproduce**:
Go to a Grafana instance (e.g. play.grafana.com). Activate your screenreader or open your dev tools and inspect the accessibility tree. Notice:

- a `header` element is not defined even though we clearly have a header
- the `main` element is marked up incorrectly (I would expect it to be on the `id="pageContent"` as that is where the skip to content link moves focus to
- on the welcome page there is an empty `section` labelled `Dashboard submenu` (if empty it should not render)
- the blog posts widget are articles with no `aria-describedby` pointing to the blog post titles.

**Expected Result**
- Should have a `header` element.
- Correctly define `main` (if you use it to anchor it is effectively the same as going to navigation so is pointless as it stands)
- Remove empty sections
- Add labels to repeated landmarks

![A screenshot of the VoiceOver rotor on the landmarks menu demonstrating the errors listed above.](https://github.com/grafana/grafana/assets/6906380/bdedcbc4-05a1-4216-96c5-1174b74701fb)

UHG Audit ID: 1.31131
